arangod/Aql/AsyncPrefetchSlotsManager.cpp

```diff
@@ -0,0 +1,114 @@
+////////////////////////////////////////////////////////////////////////////////
+/// DISCLAIMER
+///
+/// Copyright 2014-2024 ArangoDB GmbH, Cologne, Germany
+/// Copyright 2004-2014 triAGENS GmbH, Cologne, Germany
+///
+/// Licensed under the Business Source License 1.1 (the "License");
+/// you may not use this file except in compliance with the License.
+/// You may obtain a copy of the License at
+///
+///     https://github.com/arangodb/arangodb/blob/devel/LICENSE
+///
+/// Unless required by applicable law or agreed to in writing, software
+/// distributed under the License is distributed on an "AS IS" BASIS,
+/// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and
+/// limitations under the License.
+///
+/// Copyright holder is ArangoDB GmbH, Cologne, Germany
+///
+/// @author Jan Steemann
+////////////////////////////////////////////////////////////////////////////////
+
+#include "AsyncPrefetchSlotsManager.h"
+#include "Assertions/Assert.h"
+
+#include <algorithm>
+#include <utility>
+
+namespace arangodb::aql {
+
+AsyncPrefetchSlotsReservation::AsyncPrefetchSlotsReservation() noexcept
+    : _manager(nullptr), _slotsReserved(0) {}
+
+AsyncPrefetchSlotsReservation::AsyncPrefetchSlotsReservation(
+    AsyncPrefetchSlotsManager* manager, size_t slotsReserved) noexcept
+    : _manager(manager), _slotsReserved(slotsReserved) {
+  // if number of slots is > 0, we must have a manager
+  TRI_ASSERT(_slotsReserved == 0 || _manager != nullptr);
+}
```

Comment:
Can we just take the manager by reference? Then it is always guaranteed to be non-null.